Installation failure (inside docker image): `setuptools-scm` needs `git` executable if `baybe` is being installed from git

This is similar to the listing of install requirements in https://github.com/emdgroup/baybe/issues/513#issuecomment-2746346991 . Still, my situation is slightly different:

In my application, I added the following dependency to `pyproject.toml`:
```
dependencies = [
    "baybe @ git+https://github.com/emdgroup/baybe.git",
    ...
]
```

The corresponding `poetry lock` / `poetry install` calls were successful on my local development machine, but failed when building a docker image.

* `poetry lock` showed the error `setuptools-scm was unable to detect version`
* If I just used the lock file from my development machine, `poetry install` failed with the not very helpful message
```
111.1   - Installing baybe (0.12.2.post422 8428797)
------
Dockerfile:30
--------------------
  28 |     COPY pyproject.toml poetry.lock ./
  29 |     
  30 | >>> RUN POETRY_VIRTUALENVS_CREATE=false poetry install --no-cache --no-root
  31 |     
  32 |     COPY .streamlit/ .streamlit/
--------------------
ERROR: failed to solve: process "/bin/sh -c POETRY_VIRTUALENVS_CREATE=false poetry install --no-cache --no-root" did not complete successfully: exit code: 1
Error: Process completed with exit code 1.
```

After several hours of analyzing, I found out, that the issue results from the `git` executable not being available inside my docker image.

So in the end, together with (https://github.com/emdgroup/baybe/issues/513#issuecomment-2746346991), my problem was solved by adding the following line to the `Dockerfile`:
```
RUN apt-get -y update && apt-get -y install gcc g++ git
```

If possible, can this be either be documented in some kind of prominent FAQ or maybe addressed with a proper error message during installation?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -57,6 +57,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 - Incorrect optimization direction with `PSTD` with a single minimization target
+- Provide version fallback in case scm fails to infer version during installation
 
 ### Removed
 - `fuzzy_row_match` will no longer warn about entries not matching to the search space 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -173,6 +173,7 @@ build-backend = "setuptools.build_meta"
 [tool.setuptools_scm]
 version_scheme = 'post-release'
 local_scheme = "dirty-tag"
+fallback_version = "0.0+unknown"
 
 [tool.setuptools]
 packages = ["baybe"]

